src/services/ShillMessageDataService.ts

```diff
@@ -28,7 +28,19 @@ export class ShillMessageDataService {
       .all();
 
     records.forEach(record => {
-      items.push(this.shillMessage(record));
+      const content = record.get('content') as string;
+      const expiry = record.get('expiry') as string;
+      let expiredDate: number;
+      let currentDate: number;
+    
+      if (expiry) {
+        expiredDate = Date.parse(expiry);
+        currentDate = Date.now();
+      }
+      
+      if (content && (!expiry || expiredDate > currentDate)) {
```

Comment:
Something that really helps "demystify" these if-statements is turning the multi-condition checking into a variable. It's also advantageous to perform type checks instead of "coalescing truthy-ness" checks. A typeguard is preferred when not a "primitive".

**Example**
```typescript
  // Type check rather than coalescing truthy-ness checking
  if (typeof expiry === 'string') {
    // ...
  }

  // Words so that we can string them together like "prose", making our code read like a sentence
  const contentHasExpired = expiredDate > currentDate;
  const contentIsValid = typeof content === 'string';

  // Much cleaner to read
  if (contentIsValid && !contentHasExpired) {
    // ...
  }
```

This may be a bit of a contrived example but moving forward we should try to write our code as if its "prose". This will get better with time and we'll have code that is much easier to read.

Reply:
Valuable feedback! That example does seem easier to read.  I'll refactor.

Reply:
Thanks! I appreciate the warm reception and the change!